### ISSUE TITLE
Is it normal for the second in the series of calculations to slow dramatically?

### ISSUE THREAD

**Issue by Talinn92** [altf4/untwister#47]
@moloch-- @altf4 

I might be new to using this tool but I've tried to run it on Kali/Mint and every time after initially running it and going through the first calculation it does where it uses all of my CPU power to do it relatively quickly, it'll suddenly start a new series of calculation but drop my CPU usage to like 1/5th of what it should be bringing down the amount of calculations/sec from over a million to like 50,000 and take far longer to do. Is there a reason for this or is this just how it is coded? Been running Linux on VM with an AMD cpu.

![related](https://user-images.githubusercontent.com/100192042/155089733-a0e8e027-d567-42c9-b441-d3826d5b529b.PNG)

Circled the relevant part.

**Comment by moloch--** [comment]
Weird, not seen this behavior before

**Comment by Talinn92** [comment]
Yeah just tried it again and it went from like 1.2 million and about fifty minutes to 50,000 and 19 hours with the CPU dropping to basically nothing.

**Comment by Talinn92** [comment]
@moloch-- It seems like it drops to just 1 cpu thread after the first calculation for no apparent reason?

**Comment by Talinn92** [comment]
![untwisterdead](https://user-images.githubusercontent.com/100192042/155212774-2c35606a-a2f1-4c10-9d98-1b2dd77fb688.jpg)

@moloch-- @altf4

**Comment by moloch--** [comment]
very odd never seen this before

**Comment by Talinn92** [comment]
@moloch-- I sort of narrowed it down to an issue, I was using bounded prngs for ruby-rand which was added here https://github.com/hyprwired/untwister. But it doesn't seem to work with values values <100. Do you know if there is a reason for this?

**Comment by moloch--** [comment]
Not sure sorry!
